metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,558 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.Case
+import scala.meta.Defn
+import scala.meta.Pkg
+import scala.meta.Template
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta.inputs.Position
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.CodeAction
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+
+import org.eclipse.lsp4j.CodeActionParams
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.{lsp4j => l}
+
+class BracelessBracefulSwitchCodeAction(
+    trees: Trees,
+    buffers: Buffers
+) extends CodeAction {
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  override def contribute(params: CodeActionParams, token: CancelToken)(implicit
+      ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+    val maybeTree =
+      if (range.getStart == range.getEnd)
+        trees
+          .findLastEnclosingAt[Tree](
```

Comment:
```suggestion
          .findLastEnclosingAt[Template](
```

and in the next step:

```
```suggestion
          .findLastEnclosingAt[Block](
```

so this way we will no need to do all the cases for val/var etc.

Reply:
No. only class/enum/trait/object are template holders. Also, the code action should include the type it is removing/adding braces to/from in the title. Otherwise, the braces would be added to or removed from the wrong subject. I have another idea for refactoring the redundant code, which I would do at the end. It is based on just passing the right subtree arguments and title.

Reply:
We can check the parent of the template for the title and have a function for that.

Reply:
Yes! Good idea actually! But then no code action would appear when the cursor is on the class name for example. That makes it super counter intuitive!

Reply:
So just for making the code shorter, this would break a functionality! Not good!

Reply:
Maybe!